src/dataframerow/dataframerow.jl

```diff
@@ -5,7 +5,7 @@ immutable DataFrameRow{T <: AbstractDataFrame}
 end
 
 function Base.getindex(r::DataFrameRow, idx::AbstractArray)
-    return DataFrameRow(r.df[[idx]], r.row)
+    return DataFrameRow(r.df[collect(idx)], r.row)
```

Comment:
I just wanted to make a minimal modification originally, but I see your point. I removed `collect` and updated the PR. Tests run fine, so if it had a role it is not immediately apparent.